libs/shared/config-cat-core/src/lib/result-mapping.ts

```diff
@@ -0,0 +1,44 @@
+import { IEvaluationDetails } from 'configcat-js-ssr';
+import { ResolutionDetails, ResolutionReason, TypeMismatchError, StandardResolutionReasons } from '@openfeature/core';
+
+export function toResolutionDetails<T extends PrimitiveTypeName>(
+  type: T,
+  value: unknown,
+  data: Omit<IEvaluationDetails, 'value'>,
+  reason?: ResolutionReason,
+): ResolutionDetails<PrimitiveType<T>> {
+  if (!isType(type, value)) {
+    throw new TypeMismatchError(`Requested ${type} flag but the actual value is ${typeof value}`);
+  }
+
+  const matchedTargeting = 'matchedEvaluationRule' in data ? data.matchedEvaluationRule : data.matchedTargetingRule;
+  const matchedPercentage =
+    'matchedEvaluationPercentageRule' in data ? data.matchedEvaluationPercentageRule : data.matchedPercentageOption;
```

Comment:
If we do the suggested package version upgrades, we won't need to check for old property names.

```suggestion
  const matchedTargeting = data.matchedTargetingRule;
  const matchedPercentage = data.matchedPercentageOption;
```